spec/support/git_spec_helpers.rb

```diff
@@ -23,6 +23,41 @@ def repo(options = {})
     end
   end
 
+  # Creates a local repository in 'local-repo' folder with a remote repository
+  # which points to 'some-repo' folder. The current working directory is set to
+  # the local repository's root directory.
+  def with_remote_repo
+    repo do
+      touch 'update-me'
+      `echo update >> update-me`
+      touch 'delete-me'
+      `echo delete >> delete-me`
+      `git add . 2>&1 > #{File::NULL}`
+      `git commit -m "Initial commit" 2>&1 > #{File::NULL}`
+
+      Dir.chdir '..' do
+        remote_url = "file://#{File.expand_path('some-repo')}"
+        `git clone "#{remote_url}" local-repo 2>&1 > #{File::NULL}`
+      end
+
+      Dir.chdir '../local-repo' do
+        yield if block_given?
+      end
+    end
+  end
```

Comment:
Looking at the issues with this implementation of `with_remote_repo`, I'm not entirely convinced this helper is going to get you much. We should probably stick with first principles in the specs:

```ruby
remote_repo = repo do
  # Create remote repository to our liking
  ...
end

local_repo = repo do
  `git remote add origin file://#{remote_repo}`
  `git fetch origin && git reset --hard origin/master`

  # Create local repository to our liking
  ...
  # Validate expectations
end
```

Reply:
This will work. Thanks for the tip